warehouse/query-core/src/main/java/datawave/query/iterator/logic/AndIterator.java

```diff
@@ -126,29 +131,43 @@ public boolean hasNext() {
         return next != null;
     }
     
+    /**
+     * Test all layers of cache for the minimum, then if necessary advance heads
+     * 
+     * @param minimum
+     *            the minimum to return
+     * @return the first >= minimum or null if none exists
+     * @throws IllegalStateException
+     *             if prev >= minimum
+     */
     public T move(T minimum) {
         if (null == includeHeads) {
             throw new IllegalStateException("initialize() was never called");
         }
         
-        Set<T> headSet = includeHeads.keySet().headSet(minimum);
+        if (prev != null && prev.compareTo(minimum) >= 0) {
```

Comment:
If we are not allowed to move backwards, then why all of that work in the ivarator iterator?

Reply:
The work in the ivarator is to ensure it properly adheres to the seek() contract. It should never seek backwards from this point since move is forward only.

Its possible that means it never matters with the fix below